### ISSUE TITLE
Input error not being announced.

### ISSUE THREAD

**Issue by akawanishi** [jpmorganchase/salt-ds#4767]
### Latest version

- [x] I have tested the latest version

### Description

Hi we noticed that in the Input component the html <input> does not reference the id of the status indicator id as part of its aria-lablledby it only references the id of the input label, so when the focusing on the input the error is not announced. I see that the error indicator has a aria-label, but since it is a different html element it does not get read by screen reader.
Please add the id of the status indicator as part of the input aria-labelledby

### Steps to reproduce

use voice over or NVDA and focus on the input element.

### Package name(s)

Core (@salt-ds/core)

### Browser

- [x] Chrome
- [x] Safari
- [x] Firefox
- [x] Microsoft Edge

### Operating system

- [x] macOS
- [x] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [x] I am an employee of JPMorgan Chase & Co.
